clients/src/test/java/org/apache/kafka/common/metrics/MetricsGrpcClient.java

```diff
@@ -0,0 +1,154 @@
+import org.slf4j.LoggerFactory;
+import org.slf4j.Logger;
+
+import java.net.UnknownHostException;
+import java.util.concurrent.TimeUnit;
+import java.net.InetAddress;
+import java.util.List;
+
+
+public class MetricsGrpcClient implements AutoCloseable {
+    private static final Logger log = LoggerFactory.getLogger(MetricsGrpcClient.class);
+
+    public final StreamObserver<ExportMetricsServiceResponse> streamObserver =
+            new StreamObserver<ExportMetricsServiceResponse>() {
+
+        @Override
+        public void onNext(ExportMetricsServiceResponse value) {
+            // Do nothing since response is blank
+        }
+
+        @Override
+        public void onError(Throwable t) {
+            log.debug("Unable to export metrics request to {}. gRPC Connectivity in: {}:  {}",
+                    endpoint, getChannelState(), t.getCause().getMessage());
+        }
+
+        @Override
+        public void onCompleted() {
+            log.info("Successfully exported metrics request to {}", endpoint);
+        }
+    };
+
+    public static final String GRPC_HOST_CONFIG = "OTEL_EXPORTER_OTLP_ENDPOINT";
+
+    public static final int DEFAULT_GRPC_PORT = 4317;
+
+    public static final int GRPC_CHANNEL_TIMEOUT = 30;
+
+    private MetricsServiceGrpc.MetricsServiceStub grpcClient;
+
+    private ManagedChannel grpcChannel;
+
+    private String endpoint;
+
+
+    /**
+     * Start a {@link ManagedChannel} and a {@link MetricsServiceGrpc} at an endpoint defined by the
+     * {@value GRPC_HOST_CONFIG} environment variable
+     */
+    public void initialize() {
+        if (grpcChannel == null) {
+            String grpcHost = System.getenv(GRPC_HOST_CONFIG);
+
+            if (Utils.isBlank(grpcHost)) {
+                log.info("environment variable {} is not set", GRPC_HOST_CONFIG);
+
+                try {
+                    grpcHost = InetAddress.getLocalHost().getHostAddress();
+                } catch (UnknownHostException e) {
+                    log.warn("Failed to get local host address: {}", e.getMessage());
+                }
+            }
+
+            grpcChannel = ManagedChannelBuilder.forAddress(grpcHost, DEFAULT_GRPC_PORT).usePlaintext().build();
+        }
+
+        endpoint = grpcChannel.authority();
+        grpcClient = MetricsServiceGrpc.newStub(grpcChannel);
+
+        log.info("Started gRPC channel for endpoint  {}", endpoint);
+    }
+
+
+    /**
+     * Exports a ExportMetricsServiceRequest to an external endpoint
+     * @param resourceMetrics metrics to exports
+     */
+    public void export(List<ResourceMetrics> resourceMetrics) {
+        ExportMetricsServiceRequest request = ExportMetricsServiceRequest.newBuilder()
+                .addAllResourceMetrics(resourceMetrics)
+                .build();
+
+        grpcClient.export(request, streamObserver);
+    }
+
+
+    /**
+     * Shuts down the gRPC {@link ManagedChannel}
+     */
+    @Override
+    public void close()  {
+        try {
```

Comment:
Given that there was a `null` check above for the `grpcChannel` object, do we want to check for `null` here too before trying to shut down a potentially `null` object?

Reply:
yes, it could be null if the `initialize()` method hasn't been called. Updated this part.